api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxNewQueries.java

```diff
@@ -32,25 +34,47 @@
 	 * @param parameters
 	 * @return TxNew Union Query
 	 */
-	public static String getTxNewUnionQueries(int... parameters) {
-		return "select patient_id from (select patient_id, min(data_inicio) data_inicio from "
+	public static String getTxNewUnionQueries(Map<String, Integer> parameters) {
+		
+		// default to "1" so that the SQL queries will still execute when no age
+		// parameters are provided
+		String minAgeCondition = "1";
+		String maxAgeCondition = "1";
+		
+		if (parameters.get("minAge") != null) {
+			minAgeCondition = "timestampdiff(year,birthdate,data_inicio)>=" + parameters.get("minAge");
+		}
+		
+		if (parameters.get("maxAge") != null) {
+			maxAgeCondition = "timestampdiff(year,birthdate,data_inicio)<" + parameters.get("maxAge");
```

Comment:
@nthfloor shouldn't it be `maxAgeCondition = "timestampdiff(year,birthdate,data_inicio)<=" + parameters.get("maxAge");` (with **<=**)

Reply:
Thanks @ynurmahomed this was it.

To cater for the "<1" age cohort I have left it like this and instead updated the Agecohorts in the txNewDataset